src/client/client.rs

```diff
@@ -120,6 +129,23 @@ pub(crate) struct ClientOptions {
     pub(crate) connect_retry_delay: Duration,
 }
 
+impl fmt::Debug for ClientOptions {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ClientOptions")
+         .field("host", &self.host)
+         .field("port", &self.port)
+         .field("username", &self.username)
+         .field("keep_alive", &self.keep_alive)
```

Comment:
This is a great change! Can you add a comment here to say we deliberately skip `.password` to avoid leaking secrets in logs?

Reply:
Good point, just added it.